### DISCUSSION TITLE
SMTP dual listener mode 25/tcp and 587/tcp

### DISCUSSION THREAD

**Discussion by andrsharov** [postalserver/postal#1968] [Feature suggestions]
Hi, everyone.

My question about similate use ports 25 and 587. Is this possible?

1. Big Post servers(like GMAIL) use 25/tcp port for sending emails to another Big Post Servers (like Yahoo). This is standart. I think, because DNS MX record point only to 25/tcp port. 
2. Email Clients(Thunderbird, Joomla, Drupal) preffer to use 587/tcp port, because at the most cloud providers 25/tcp port are blocked. This is also standart.

How to setup postal to listen both port(25,587) simultaneously (or third(25,2525,587) ports simultaneously)?

But these port are needed for different purposes:
1. Only port 25/tcp needed as relay port (for incoming emails from another servers). 
2. Ports 465,587 or any random ports needed only for sending email (from drupal, joomla, and any another email client)

I try to setup my postal.yml file, like this:

```
smtp_server:
  tls_enabled: true
  port: 25,587
```
or like this:
```
smtp_server:
  tls_enabled: true
  port: 
    - 25
    - 587
```
But it don't working.
Also I found in file https://github.com/postalserver/postal/blob/main/config/postal.defaults.yml 
this section:
```
smtp_relays:
  -
    hostname:
    port: 25
    ssl_mode: Auto
```
I tried to configure it, but without success.  `ss -ltn` shows than only one listener is worked ( on port 587/tcp) 
```
root@mail:~# ss -ltn
State                           Recv-Q                           Send-Q                                                     Local Address:Port                                                     Peer Address:Port                          Process                          
LISTEN                          0                                1024                                                           127.0.0.1:5000                                                          0.0.0.0:*                                                              
LISTEN                          0                                4096                                                           127.0.0.1:783                                                           0.0.0.0:*                                                              
LISTEN                          0                                511                                                              0.0.0.0:80                                                            0.0.0.0:*                                                              
LISTEN                          0                                4096                                                       127.0.0.53%lo:53                                                            0.0.0.0:*                                                              
LISTEN                          0                                128                                                              0.0.0.0:22                                                            0.0.0.0:*                                                              
LISTEN                          0                                511                                                              0.0.0.0:443                                                           0.0.0.0:*                                                              
LISTEN                          0                                128                                                                    *:587                                                                 *:*                                                              
LISTEN                          0                                128                                                                 [::]:22                                                               [::]:*      
```
Why I do need this? I need than unsubscribe emails are working (they are only can works if I will be used 25/tcp port) and email clients on alternate port 587/tcp are working to simultaneously. 
Maybe I badly read instruction?

**Comment by willpower232** [comment]
Postal was not designed to be used with email client software so it only needs port 25 and does not have the code to listen on multiple ports at the same time.

If you wanted to, you could use iptables or something similar to forward traffic for both ports so that both were "open" but this shouldn't be necessary.

`smtp_relays` are completely unrelated, I think they are for outgoing traffic only

**Reply by dragoangel** [reply]
Want remind you about https://github.com/postalserver/postal/discussions/1517 discussion
